Source/WebCore/platform/spiel/PlatformSpeechSynthesizerSpiel.cpp

```diff
@@ -0,0 +1,258 @@
+    if (!audioSink) {
+        GST_ERROR("Failed to create GStreamer audio sink element");
+        return;
+    }
+
+    // Async API can't really be used here because the voices initialization
+    // (PlatformSpeechSynthesizer::initializeVoiceList()) is synchronous.
+    GUniqueOutPtr<GError> error;
+    m_speaker = adoptGRef(spiel_speaker_new_sync(nullptr, &error.outPtr()));
+    if (error) {
+        WTFLogAlways("Spiel speaker failed to initialize: %s", error->message);
+        return;
+    }
+
+    g_object_set(m_speaker.get(), "sink", audioSink.get(), nullptr);
+
+    // TODO: Plumb support for boundaryEventOccurred? Using range-started signal?
+
+    g_signal_connect_swapped(m_speaker.get(), "utterance-started", G_CALLBACK(+[](SpielSpeechWrapper* self, SpielUtterance*) {
+        self->m_platformSynthesizer.client().didStartSpeaking(*self->m_utterance);
+    }), this);
+
+    g_signal_connect_swapped(m_speaker.get(), "utterance-finished", G_CALLBACK(+[](SpielSpeechWrapper* self, SpielUtterance*) {
+        self->m_platformSynthesizer.client().didFinishSpeaking(*self->m_utterance);
+    }), this);
+
+    g_signal_connect_swapped(m_speaker.get(), "utterance-canceled", G_CALLBACK(+[](SpielSpeechWrapper* self, SpielUtterance*) {
+        self->m_platformSynthesizer.client().didFinishSpeaking(*self->m_utterance);
+    }), this);
+
+    g_signal_connect_swapped(m_speaker.get(), "utterance-error", G_CALLBACK(+[](SpielSpeechWrapper* self, SpielUtterance*) {
+        self->m_platformSynthesizer.client().speakingErrorOccurred(*self->m_utterance);
+    }), this);
+
+    g_signal_connect_swapped(m_speaker.get(), "notify::paused", G_CALLBACK(+[](SpielSpeechWrapper* self, SpielSpeaker* speaker) {
+        gboolean isPaused;
+        g_object_get(speaker, "paused", &isPaused, nullptr);
+        if (isPaused)
+            self->m_platformSynthesizer.client().didPauseSpeaking(*self->m_utterance);
+        else
+            self->m_platformSynthesizer.client().didResumeSpeaking(*self->m_utterance);
+    }), this);
+
+    g_signal_connect_swapped(m_speaker.get(), "notify::voices", G_CALLBACK(+[](SpielSpeechWrapper* self, SpielSpeaker*) {
+        self->m_platformSynthesizer.client().voicesDidChange();
+    }), this);
+}
+
+SpielSpeechWrapper::~SpielSpeechWrapper()
+{
+    g_signal_handlers_disconnect_by_data(m_speaker.get(), this);
+}
+
+Vector<RefPtr<PlatformSpeechSynthesisVoice>> SpielSpeechWrapper::initializeVoiceList()
+{
+    Vector<RefPtr<PlatformSpeechSynthesisVoice>> platformVoices;
+    auto voices = spiel_speaker_get_voices(m_speaker.get());
+    unsigned position = 0;
+    m_voices.clear();
+    while (auto item = g_list_model_get_item(voices, position)) {
+        auto voice = SPIEL_VOICE(item);
+        const char* const* languages = spiel_voice_get_languages(voice);
+        if (!G_N_ELEMENTS(languages))
+            continue;
+
+        auto language = makeString(span(languages[0]));
```

Comment:
Ideally, you would make a web voice for every language in a spiel voice. If you don't then the web will only see a single Afrikans eSpeak voice, for example. Instead of the ~130 other languages

Reply:
Ah yes, good point!